src/ImageGallery.jsx

```diff
@@ -652,16 +654,17 @@ export default class ImageGallery extends React.Component {
     const CENTER = 'center';
     const RIGHT = 'right';
 
-    switch (index) {
-      case (currentIndex - 1):
-        alignment = ` ${LEFT}`;
-        break;
-      case (currentIndex):
-        alignment = ` ${CENTER}`;
-        break;
-      case (currentIndex + 1):
-        alignment = ` ${RIGHT}`;
-        break;
+    if (index === currentIndex - 1) {
+      alignment = ` ${LEFT}`;
+    } else if (index === currentIndex) {
+      alignment = ` ${CENTER}`;
+    } else if (index === currentIndex + 1) {
+      alignment = ` ${RIGHT}`;
+    } else if (this.props.preLoad > 0) {
+      const preLoadIndex = Math.max(index - currentIndex, 0)
```

Comment:
although it's admittedly cool to use `Math.max`, I don't think you need it, since you're already checking that `preLoadIndex > 0` on the next line. This should be just fine:

```suggestion
      const preLoadIndex = index - currentIndex
```

Reply:
Thanks, good catch.